[WINP] Fix kuttl test parallel 055

ciiay opened the pull request:
Signed off: @ciiay 

**What type of PR is this?**
/kind failing-test

**What does this PR do / why we need it**:
Remove acceptance pipeline blockers in order to enable auto-release.

**Which issue(s) this PR fixes**:

Fixes [GITOPS-2857](https://issues.redhat.com/browse/GITOPS-2857)

**Test acceptance criteria**:

* [ ] Unit Test
* [x] E2E Test

**How to test changes / Special notes to the reviewer**:
1. Pull [operator-e2e](https://gitlab.cee.redhat.com/gitops/operator-e2e) repo to local and go to `gitops-operator` directory.
2. Apply changes from this PR
3. Run `kubectl kuttl test ./tests/parallel --config ./tests/parallel/kuttl-test.yaml --test 1-055_validate_notification_controller` command to confirm this test passes.

Output from my end:
```
➜  gitops-operator git:(master) ✗ kubectl kuttl test ./tests/parallel --config ./tests/parallel/kuttl-test.yaml --test 1-055_validate_notification_controller
2023/06/01 16:26:17 kutt-test config testdirs is overridden with args: [ ./tests/parallel ]
=== RUN   kuttl
    harness.go:462: starting setup
    harness.go:252: running tests using configured kubeconfig.
I0601 16:26:18.304116   93658 request.go:682] Waited for 1.048364333s due to client-side throttling, not priority and fairness, request: GET:https://api.rhoms-4.12-060104.dev.openshiftappsvc.org:6443/apis/autoscaling/v1?timeout=32s
    harness.go:275: Successful connection to cluster at: https://api.rhoms-4.12-060104.dev.openshiftappsvc.org:6443
I0601 16:26:28.306437   93658 request.go:682] Waited for 2.094510083s due to client-side throttling, not priority and fairness, request: GET:https://api.rhoms-4.12-060104.dev.openshiftappsvc.org:6443/apis/monitoring.coreos.com/v1beta1?timeout=32s
    harness.go:360: running tests
    harness.go:73: going to run test suite with timeout of 1200 seconds for each step
    harness.go:372: testsuite: ./tests/parallel has 68 tests
=== RUN   kuttl/harness
=== RUN   kuttl/harness/1-055_validate_notification_controller
=== PAUSE kuttl/harness/1-055_validate_notification_controller
=== CONT  kuttl/harness/1-055_validate_notification_controller
    logger.go:42: 16:26:30 | 1-055_validate_notification_controller | Ignoring check.yaml as it does not match file name regexp: ^(\d+)-(?:[^\.]+)(?:\.yaml)?$
    logger.go:42: 16:26:30 | 1-055_validate_notification_controller | Creating namespace: kuttl-test-huge-pony
    logger.go:42: 16:26:30 | 1-055_validate_notification_controller/1-install | starting test step 1-install
    logger.go:42: 16:26:35 | 1-055_validate_notification_controller/1-install | ArgoCD:kuttl-test-huge-pony/example-argocd created
    logger.go:42: 16:26:36 | 1-055_validate_notification_controller/1-install | test step completed 1-install
    logger.go:42: 16:26:36 | 1-055_validate_notification_controller/2-enable_notification | starting test step 2-enable_notification
I0601 16:26:38.320936   93658 request.go:682] Waited for 1.70655s due to client-side throttling, not priority and fairness, request: GET:https://api.rhoms-4.12-060104.dev.openshiftappsvc.org:6443/apis/migration.k8s.io/v1alpha1?timeout=32s
    logger.go:42: 16:26:41 | 1-055_validate_notification_controller/2-enable_notification | ArgoCD:kuttl-test-huge-pony/example-argocd updated
    logger.go:42: 16:26:45 | 1-055_validate_notification_controller/2-enable_notification | test step completed 2-enable_notification
    logger.go:42: 16:26:45 | 1-055_validate_notification_controller/3-disable_notification | starting test step 3-disable_notification
I0601 16:26:48.342740   93658 request.go:682] Waited for 3.045766875s due to client-side throttling, not priority and fairness, request: GET:https://api.rhoms-4.12-060104.dev.openshiftappsvc.org:6443/apis/psmdb.percona.com/v1-11-0?timeout=32s
    logger.go:42: 16:26:49 | 1-055_validate_notification_controller/3-disable_notification | ArgoCD:kuttl-test-huge-pony/example-argocd updated
    logger.go:42: 16:26:50 | 1-055_validate_notification_controller/3-disable_notification | test step completed 3-disable_notification
    logger.go:42: 16:26:50 | 1-055_validate_notification_controller | skipping kubernetes event logging
    logger.go:42: 16:26:50 | 1-055_validate_notification_controller | Deleting namespace: kuttl-test-huge-pony
=== CONT  kuttl
    harness.go:405: run tests finished
    harness.go:513: cleaning up
    harness.go:570: removing temp folder: ""
--- PASS: kuttl (40.65s)
    --- PASS: kuttl/harness (0.00s)
  ```

ciiay commented:
@varshab1210  I see the change made in PR #490 , but there's an error from the log says
```
=== CONT  kuttl/harness/1-055_validate_notification_controller
    logger.go:42: 10:43:01 | 1-055_validate_notification_controller | Ignoring errors.yaml as it does not match file name regexp: ^(\d+)-(?:[^\.]+)(?:\.yaml)?$
```
So I renamed the error file back to its original format which is `03-error.yaml` so it's correctly read by harness. I figured the whole point of adding this `check.yaml` file is to increase the timeout, but the test seems very simple and for local test it only takes a few seconds to finish the 3 steps after changing the error.yaml file name back to `03-error.yaml`. Let me know if I was wrong and we can discuss this further.

ciiay commented:
/retest-required

ciiay commented:
/retest

ciiay commented:
Applied this fix in PR #481 and it passed all parallel check in there too.

varshab1210 commented on test/openshift/e2e/parallel/1-055_validate_notification_controller/03-errors.yaml:
```diff
@@ -27,10 +34,3 @@ kind: RoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: example-argocd-argocd-notifications-controller
----
```
@ciiay Any reason behind placing this assertion on the top?

ciiay replied:
Will remove it.

varshab1210 commented:
> @varshab1210 I see the change made in PR #490 , but there's an error from the [log](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/redhat-developer_gitops-operator/490/pull-ci-redhat-developer-gitops-operator-master-v4.12-kuttl-parallel/1650809757851717632/build-log.txt) says
> 
> ```
> === CONT  kuttl/harness/1-055_validate_notification_controller
>     logger.go:42: 10:43:01 | 1-055_validate_notification_controller | Ignoring errors.yaml as it does not match file name regexp: ^(\d+)-(?:[^\.]+)(?:\.yaml)?$
> ```
> 
> So I renamed the error file back to its original format which is `03-error.yaml` so it's correctly read by harness. I figured the whole point of adding this `check.yaml` file is to increase the timeout, but the test seems very simple and for local test it only takes a few seconds to finish the 3 steps after changing the error.yaml file name back to `03-error.yaml`. Let me know if I was wrong and we can discuss this further.

Hi @ciiay, as you know the whole point of adding a check.yaml was to increase the timeout which means this test failure existed even before this change. This test had failed more than 15 times across this repository (one of the most failed test cases earlier) and now  failure has decreased because of this change only. There was no configuration mistake with not numbering the errors.yaml, its just kuttl's way of saying that it would not consider the file during its usual execution. I don't think changing this test back to how it used to be would solve this problem as the issue existed then too and as I said, its difficult reproduce this locally.

ciiay commented:
@varshab1210 When you put `check.yaml` the `errors.yaml` file didn't get read by harness. Both this PR and #481 PR I applied this change and they passed this parallel test in any OCP version for the first time. This is an approval that not only it passed local tests but also CI check. I don't know other ways I could test/reproduce this issue. Any suggestions?

varshab1210 commented:
Hi @ciiay 

>  When you put check.yaml the errors.yaml file didn't get read by harness

If that was the case, why does the error say 

```
case.go:361: failed in step 4-check    
case.go:363: resource matched of kind: apps/v1, Kind=Deployment
case.go:363: resource matched of kind: /v1, Kind=Secret
case.go:363: resource matched of kind: /v1, Kind=ConfigMap
case.go:363: resource matched of kind: rbac.authorization.k8s.io/v1, Kind=RoleBinding
```
The statement **resource matched of kind:** means that the resource which was not expected to be on the cluster has been found. The above resources are part of [errors.yaml](https://github.com/redhat-developer/gitops-operator/blob/master/test/openshift/e2e/parallel/1-055_validate_notification_controller/errors.yaml) which have been checked as part of 04-check.yaml. Why does it points to four resources only? This means that the other resources (ArgoCD, ServiceAccount and Role) have already been removed and the assertion matched.

> This is an approval that not only it passed local tests but also CI check

This test has never been consistently failing, its a flaky test. Take a look at the [flake board](https://github.com/anandrkskd/gitops-operator-flake-board/blob/main/output/flake-stats.md), we have around 25 PRs in this repository but this has failed in only a few. If the implementation was wrong, you would have observed this test failing for all the PRs. Try to run the same test locally (which includes 04-check.yaml) and check if the test passes then.

We can merge this PR and see if it helps but I am pretty sure it won't as this is how the test has always been.

ciiay commented:
As discussed with @varshab1210 , removed `check.yaml` as the timeout is configured in `kuttl-test.yaml` already. Instead, I added sleep before disable notification because sometimes it takes longer for the argocd-controller pod to spin up, which is watching on the argocd manifest changes. If the pod is not ready then it will not realize the `notifications: false` change in argocd manifest, then it won't start reconciling which causes the test failure. I have tried sleep 5s, 10s and 15s, and 15s is long enough for argocd-controller to spin up.

I tested this change locally 20 times in a row and every time it passed.
```
--- PASS: kuttl (48.58s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (40.27s)
PASS

--- PASS: kuttl (46.34s)
    --- PASS: kuttl/harness (0.01s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (38.14s)
PASS

--- PASS: kuttl (47.94s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (39.71s)
PASS

--- PASS: kuttl (46.51s)
    --- PASS: kuttl/harness (0.01s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (38.22s)
PASS

--- PASS: kuttl (47.77s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (39.56s)
PASS

--- PASS: kuttl (48.78s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (40.46s)
PASS

--- PASS: kuttl (50.95s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (42.69s)
PASS

--- PASS: kuttl (48.25s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (39.98s)
PASS

--- PASS: kuttl (49.72s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (41.42s)
PASS

--- PASS: kuttl (47.37s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (39.10s)
PASS

--- PASS: kuttl (47.61s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (39.32s)
PASS

--- PASS: kuttl (46.07s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (37.79s)
PASS

--- PASS: kuttl (50.84s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (42.53s)
PASS

--- PASS: kuttl (46.54s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (38.32s)
PASS

--- PASS: kuttl (47.67s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (39.34s)
PASS

--- PASS: kuttl (48.60s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (40.28s)
PASS

--- PASS: kuttl (47.60s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (39.26s)
PASS

--- PASS: kuttl (52.33s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (44.08s)
PASS

--- PASS: kuttl (49.52s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (41.26s)
PASS

--- PASS: kuttl (50.77s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/1-055_validate_notification_controller (42.58s)
PASS
```

<img width="580" alt="Screenshot 2023-06-02 at 3 33 55 PM" src="https://github.com/redhat-developer/gitops-operator/assets/26255262/4bdac0f5-cf79-4d11-821f-ddc6906fa42a">

varshab1210 commented:
Thanks @ciiay 
I would recommend to add a 03-assert.yaml instead of sleep. Something like

```
apiVersion: v1
kind: Pod
metadata:
  labels:
    app.kubernetes.io/name: argocd-application-controller
status:
  phase: Running
```

varshab1210 commented:
Closing this, please reopen if required
